### PR TITLE
Enable RuboCop Performance linters

### PR DESCRIPTION
**Pull request by kaylareopelle**
# Overview

This PR turns on RuboCop Performance linters that are compatible with the Rubies we support and represent a very small amount of effort to resolve. Linters incompatible with the Rubies we support have been marked with TODOs and the minimum supported Ruby version.

Submitter Checklist:
- [ ] Include a link to the related GitHub issue, if applicable
- [ ] Include a security review link, if applicable

# Testing
The agent includes a suite of unit and functional tests which should be used to
verify your changes don't break existing functionality. These tests will run with 
Github Actions when a pull request is made. More details on running the tests locally can be found 
[here for our unit tests](https://github.com/newrelic/newrelic-ruby-agent/blob/main/test/README.md), 
and [here for our functional tests](https://github.com/newrelic/newrelic-ruby-agent/blob/main/test/multiverse/README.md).
For most contributions it is strongly recommended to add additional tests which
exercise your changes. 

# Reviewer Checklist
- [ ] Perform code review
- [ ] Add performance label
- [ ] Perform appropriate level of performance testing
- [ ] Confirm all checks passed
- [ ] Add version label prior to acceptance

### CODE REVIEW

File: .github/workflows/performance_tests.yml [hunk 1]
```diff
@@ -2,8 +2,6 @@ name: Performance Tests
 
 on:
   pull_request:
-    branches:
-      - main
   schedule:
```
Comment (kaylareopelle) [inline .github/workflows/performance_tests.yml:5]: TODO: Revert this change before merge.

File: bin/nrdebug [hunk 2]
```diff
@@ -97,11 +97,11 @@ class LinuxProcessDataProvider < ProcessDataProvider
   end
 
   def procline
-    File.read(proc_path('cmdline')).gsub("\000", " ")
+    File.read(proc_path('cmdline')).tr("\000", " ")
```
Comment (kaylareopelle) [inline bin/nrdebug:100]: https://ruby-doc.org/core-3.1.2/String.html#method-i-tr

**Comment by kaylareopelle** [comment]
Our perf test results remained pretty stable with these changes: https://files.slack.com/files-pri/T018FFWTM3K-F043X43C9B4/performance_results.txt (Internal)

**Review by fallwith — APPROVED** [review]
Thank you for these changes. I especially love the Ruby version annotations you've added to `.rubocop.yml`.
